src/Parallel/Printf.hpp

```diff
@@ -0,0 +1,76 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines Parallel::printf for writing to stdout
+
+#pragma once
+
+#include <charm++.h>
+#include <sstream>
+#include <string>
+#include <type_traits>
+
+namespace Parallel {
+namespace detail {
+// Print pointers
+template <typename T,
+          typename std::enable_if_t<
+              std::is_fundamental<std::decay_t<
+                  std::remove_pointer_t<std::decay_t<T>>>>::value or
+              std::is_pointer<T>::value> * = nullptr>
```

Comment:
So the second clause prints the pointer for any pointer, and the first clause prints either the pointer to a fundamental type, or the fundamental type; so why do I need the remove_pointer ?